backend/controllers/LocationsController.ts

```diff
@@ -93,6 +94,31 @@ const LocationsController = {
       return res.status(500).json({ error: err.message });
     }
   },
+
+  async addLocationComments(req: Request, res: Response) {
+    try {
+      const { id,comment } = req.body;
+      const commentProperties = ['author','text','likes','dislikes','createdAt','updatedAt'];
+      let fullComment:boolean = true;
+      const location = await Location.findById(id);
+      if ( !location||!location.comments ) {
+        return res.status(400).json({ error: "Location or comments doesn't exist" });
```

Comment:
use this to check the existence of the document and its comment field in it
await Location.exists({
- _id: id,
- comments: { $exists: true }
})

Reply:
and get rid of location variable